mongoose/messages/group-schema.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * @file Implements mongoose schema
+ * for the groups collection
+ */
+
+import mongoose, {Schema} from "mongoose";
+import Group from "../../models/messages/group";
+
+const GroupSchema = new mongoose.Schema<Group>({
+    groupName: {type: String, required: true},
+    users: {
+        user: {type: Schema.Types.ObjectId, ref: "UserModel", required: true},
```

Comment:
Shouldn't this be an array? [Mongoose docs](https://mongoosejs.com/docs/schematypes.html#arrays)

Reply:
Yes, I forgot to put the outside brackets. 

Also, I hadn't seen the syntax "users: [User]" before.  I guess it could also just be User[], no? The one I found seems to be for attributes that don't have their own object type and the objects in the list each have many attributes.

Reply:
I added brackets to my version which I think might work. I also added the version based on the docs, but I'm not sure it's right as I had to add:

const UserSchema = new Schema({ name: String });

I don't really understand the {name: String} part.